app/code/Magento/Elasticsearch/Model/Adapter/BatchDataMapper/ProductDataMapper.php

```diff
@@ -395,6 +398,14 @@ private function getAttributeOptions(Attribute $attribute, int $storeId): array
              * $attribute->getOptions() loads options into data objects which can be costly.
              */
             $options = $attribute->usesSource() ? $attribute->setStoreId($storeId)->getSource()->getAllOptions() : [];
+
+            if (!empty($options)) {
+                $options = array_combine(
+                    array_column($options, 'value'),
+                    $options
+                );
+            }
+
```

Comment:
I believe we should revert this change as well. It doesn't make sense together with foreach
```suggestion
```